Sources/Controllers/MessagesViewController+Keyboard.swift

```diff
@@ -81,23 +107,35 @@ extension MessagesViewController {
         }
     }
 
-    // MARK: - Helpers
+    private func requiredScrollViewBottomInset(forKeyboardFrame keyboardFrame: CGRect) -> CGFloat {
+        // we only need to adjust for the part of the keyboard that covers (i.e. intersects) our collection view;
+        // see https://developer.apple.com/videos/play/wwdc2017/242/ for more details
+        let intersection = messagesCollectionView.frame.intersection(keyboardFrame)
+        
+        if intersection.isNull || intersection.maxY < messagesCollectionView.frame.maxY - coordinatePrecision /* never compare floats without a tolerance */ {
```

Comment:
`intersection.isNull` has take the precision into consideration. `intersection.maxY < messagesCollectionView.frame.maxY - coordinatePrecision` is not needed.